### PR TITLE
Update to latest hie-bios

### PR DESCRIPTION
**Pull request by fendor**
Updates ghcide to use the latest hie-bios version 0.4.0

### CODE REVIEW

**Comment by cocreature** [comment]
I tested this as part of #361 and it seemed to work fine.

**Comment by fendor** [comment]
What OS does CI run? E.g. I can not rely that it builds on windows only because CI is green, right?

**Comment by cocreature** [comment]
CI only runs Linux but if `hie-bios` compiles on Windows in general I’m not too worried about that. We don’t have any windows-specific code in this part. And if you are worried about actually running rather than just compiling, our test suite isn’t going to be particularly helpful. We don’t test the hie-bios setup. All our tests just rely on `base` and not much else.

**Comment by fendor** [comment]
I remember that I had to add an additional dependency for windows on hie. I will test it on my system later.

**Comment by fendor** [comment]
Tested to build `stack84.yaml` on windows and it succeeded.

File: exe/Main.hs [hunk 1]
```diff
@@ -15,6 +15,7 @@ import Control.Exception
 import Control.Monad.Extra
 import Control.Monad.IO.Class
 import Data.Default
+import Data.Void (Void)
```
Comment (ndmitchell) [inline exe/Main.hs]: Generally no need to specify imports.

File: exe/Main.hs [hunk 2]
```diff
@@ -175,7 +176,7 @@ showEvent lock (EventFileDiagnostics (toNormalizedFilePath -> file) diags) =
 showEvent lock e = withLock lock $ print e
 
 
-cradleToSession :: Cradle -> IO HscEnvEq
+cradleToSession :: Cradle Void -> IO HscEnvEq
```
Comment (ndmitchell) [inline exe/Main.hs]: Could this equally be `Cradle a`, so it was polymorphic in the type of Cradle it receives? That seems slightly simpler.

File: stack88.yaml [hunk 3]
```diff
@@ -3,6 +3,7 @@ packages:
 - .
 extra-deps:
 - fuzzy-0.1.0.0
+- hie-bios-0.4.0
```
Comment (ndmitchell) [inline stack88.yaml]: Note this will clash with my changes in #400 - but I'm happy to rebase after this lands.

File: exe/Main.hs (hunk 2 above)
Comment (fendor) [reply exe/Main.hs]: Yeah sure, should not be a problem.

**Pull request merged**